File: 1-two-sum/two-sum.cpp
class Solution {
public:
    
    vector<int> twoSum(vector<int>& nums, int target) {

        unordered_map<int, int> mp;
        for (int i = 0; i < nums.size(); i++) {
            mp[nums[i]] = i;
        }

        for (int i = 0; i < nums.size(); i++) {
            if (i == mp[target - nums[i]]) {
                continue;
            }
            // cout << arr[mp[target - arr[i]]] << endl;
            if (nums[mp[target - nums[i]]] + nums[i] == target) {
                return {i,mp[target-nums[i]]};
            }
        }

        return {-1, -1};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer,Integer> mpp = new HashMap<>();

        for(int i = 0 ; i < nums.length ; i++){
            mpp.put(nums[i] , i);
        }

        int[] ans = new int[2];

        for(int i = 0 ; i < nums.length ; i++){
            int complement = target - nums[i];

            if(mpp.containsKey(complement) && mpp.get(complement) != i){
                ans[0] = i;
                ans[1] = mpp.get(complement);
                return ans ;
            }

        }

        return ans ;

    }
}